manticore/seth.py

```diff
@@ -147,20 +103,114 @@ def make_function_arguments(*args):
         
     @staticmethod
     def make_function_call(method_name, *args):
-        function_id = ManticoreEVM.make_function_id(method_name)
+        function_id = ABI.make_function_id(method_name)
         def check_bitsize(value, size):
             if isinstance(value, BitVec):
                 return value.size==size
             return (value & ~((1<<size)-1)) == 0
         assert len(function_id) == 4
         result = [tuple(function_id)]
-        result.append(ManticoreEVM.make_function_arguments(*args))
+        result.append(ABI.make_function_arguments(*args))
         return reduce(lambda x,y: x+y, result)
 
+
+class EVMAccount(object):
+    ''' An EVM account '''
+    def __init__(self, address, seth=None, default_caller=None):
+        ''' Encapsulates an account. 
+
+            :param address: the address of this account
+            :type address: 160 bit long integer
+            :param seth: the controlling manticore
+            :param default_caller: the default caller address for any transaction
+
+        '''
+        self._default_caller = default_caller
+        self._seth=seth
+        self._address=address
+        self._hashes = {}
+
+        if self._seth:
+            name, source_code, init_bytecode, metadata, metadata_runtime, hashes = self._seth.context['seth']['metadata'][address]
+            for signature in hashes.keys():
+                func_name = str(signature.split('(')[0])
+                self._hashes[func_name] = signature, hashes[signature]
+
+    def __int__(self):
+        return self._address
+
+    def __getattribute__(self, name):
+        ''' If this is a contract account of which we know the functions hashes
+            this will build the transaction for the function call.
+
+            Example use::
+        
+                #call funtion `add` on contract_account with argument `1000`
+                contract_account.add(1000)
+         
+        '''
+        if not name.startswith('_') and name in self._hashes.keys():
+            def f(*args, **kwargs):
+                caller = kwargs.get('caller', None)
+                value = kwargs.get('value', 0)
+                tx_data = ABI.make_function_call(str(self._hashes[name][0]), *args)
+                if caller is not None:
+                    caller = int(caller)
+                else:
+                    caller = self._default_caller
+                self._seth.transaction(caller=caller,
+                                        address=self._address,
+                                        value=value,
+                                        data=tx_data
+                                     )
+                self._caller = None
+                self._value = 0
+            return f
+        else:
+            return object.__getattribute__(self, name)            
+
+
+
+class ManticoreEVM(Manticore):
+    ''' Manticore EVM manager
+    
+        Usage Ex::
+
+            from seth import ManticoreEVM, ABI
```

Comment:
this should from `from manticore.seth`

we need to be _very_ careful with these inline code blocks since they are not tested! it really sucks to encounter broken code snippets in documentation.